### ISSUE TITLE
Cannot compile under Linux

### ISSUE THREAD

**Issue by serpheroth** [RippeR37/GLUL#29]
Hi, I tried to compile the library under Linux, but I got a bunch of compilation errors. Is there any plan to support other platforms like Linux and Mac OS X?

**Comment by RippeR37** [comment]
@serpheroth Which compiler did you use? Which version? What linux distro?

Currently i'm without linux, but i'll try to download i.e. Ubuntu image and virtual machine today and check it out, maybe even add makefile or something like that.

As for support for Linux platform - my code doesn't use anything platform-specific, only GLFW/GLEW/GLM/OGL3.3/C++11, so theoretically it should work without problems on Linux and other platforms. But in practice there are sometimes few differences etc. though i didn't test it on linux yet.

Provide more informations and i'll look into it.

**Comment by serpheroth** [comment]
I used g++ 4.8 with -std=c++11 flag. The Linux distro is Linux Mint 17.1

**Comment by RippeR37** [comment]
Well, i did check it out. It seems that most of error are connected with abscense of `throw()` or `throw(Util::Exception::FatalError)` in .cpp files which are not needed/ignored in Visual Studio.
There were other problems:
- lack of support for std::put_time (it's GCC thing, GCC 5.0 will support it finally)
- misspelled `util` instead of `Util` in one place
- other small error that pass in VS but on GCC won't.

So there are no linux-specific problems per-se, but more like lack of GCC support. I'll mark this issue and fix it ASAP on Windows with GCC 4.9 and then check if it works on Linux.

I may also provide some simple Makefile file to help build library, though it's very simple (install GLEW/GLFW/GLM), link to: `-lGL -lGLU -lGLEW -lglfw3` and set include directiories for these libraries + `include` folder from this repo. Then you only have to build .o files from .cpp files and then link it to library file or something.

**Comment by serpheroth** [comment]
Good. I will try to compile the code after your fix. I will also try to compile under Mac OS.
I tried g++-4.9, there is another problem not in your list,

```
include/Utils/File.h:16:50: error: underlying type ‘std::ios_base::openmode {aka std::_Ios_Openmode}’ of ‘Util::File::ReadMode’ must be an integral type
             enum class ReadMode : std::ios_base::openmode {
```

**Comment by RippeR37** [comment]
@serpheroth - i fixed errors (**it's on `develop` branch!**) that i found on my GCC 4.9.2 (it's from MinGW-w64 and i recommend it, becouse there were some bugs with MinGW's version with `std::sto*(...)` functions.

Also there still is one error left _Logger.cpp_ file with usage of (not-yet implemented in GCC) function `std::put_time(...)` which is already implemented in VS's compiler. This function is marked as fixed and it's support will be implemented in GCC 5.0, so until then you can comment given line and your logs will be left woithout date and time (i guess you can live with it).

There will be another patch to fix GCC warnings (it seems that GCC has much better warnings then VS's compiler), but for now it compiles. I didnt test it yet under Linux, but it looked like i fixed all error that occured earlier on Ubuntu.
